ps_democqrshooksusage.php

```diff
@@ -0,0 +1,337 @@
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+use DemoCQRSHooksUsage\Domain\Reviewer\Command\UpdateIsAllowedToReviewCommand;
+use DemoCQRSHooksUsage\Domain\Reviewer\Exception\CannotCreateReviewerException;
+use DemoCQRSHooksUsage\Domain\Reviewer\Exception\CannotToggleAllowedToReviewStatusException;
+use DemoCQRSHooksUsage\Domain\Reviewer\Exception\ReviewerException;
+use DemoCQRSHooksUsage\Domain\Reviewer\Query\GetReviewerSettingsForForm;
+use DemoCQRSHooksUsage\Domain\Reviewer\QueryResult\ReviewerSettingsForForm;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Exception\CustomerException;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ToggleColumn;
+use PrestaShop\PrestaShop\Core\Grid\Definition\GridDefinitionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Filter\Filter;
+use PrestaShop\PrestaShop\Core\Search\Filters\CustomerFilters;
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use PrestaShopBundle\Form\Admin\Type\YesAndNoChoiceType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+
+//todo: demonstrate how include custom js extensions for existing grids maybe?.
+//todo: not a single translation works for this module
+/**
+ * Class Ps_DemoCQRSHooksUsage demonstrates the usage of CQRS and hooks.
+ */
+class Ps_DemoCQRSHooksUsage extends Module
+{
+    public function __construct()
+    {
+        $this->name = 'ps_democqrshooksusage';
+        $this->version = '1.0.0';
+        $this->author = 'Tomas Ilginis';
+        $this->need_instance = 0;
+
+        parent::__construct();
+
+        $this->displayName = $this->getTranslator()->trans(
+            'Demo for CQRS and hooks usage',
+            [],
+            'Modules.Ps_DemoCQRSHooksUsage'
+        );
+
+        $this->description =
+            $this->getTranslator()->trans(
+                'Help developers to understand how to create module using new hooks and apply best practices when using CQRS',
+                [],
+                'Modules.Ps_DemoCQRSHooksUsage'
+            );
+
+        $this->ps_versions_compliancy = [
+            'min' => '1.7.6.0',
+            'max' => _PS_VERSION_,
+        ];
+    }
+
+    /**
+     * Install module and register hooks to allow grid modification.
+     *
+     * @return bool
+     */
+    public function install()
+    {
+        return parent::install() &&
```

Comment:
We could add a link to the doc that talks about the new hooks in grid and identifiable object forms